geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/IngestCommand.scala

```diff
@@ -238,47 +238,105 @@ object IngestCommand extends LazyLogging {
-      }
+      var open: () => InputStream = null
       val opened = ListBuffer.empty[CloseableIterator[InputStream]]
-      def open(): InputStream = {
-        val streams = file.open.map(_._2)
-        opened += streams
-        if (streams.hasNext) { streams.next } else {
+      var file: FileHandle = null
+
+      // try to infer the schema from stdin
+      if (inputs == Inputs.StdInInputs) {
+        Command.user.info("No converter defined - will attempt to detect schema from stdin")
+
+        // Cache the bytes from the input stream, because we'll need them again for the ingest
+        val inputBytes = System.in.readAllBytes()
+
+        // Function that returns an input stream containing the data from stdin
+        def openStdin(): InputStream = {
+          try {
+            // This prevents the schema and type inference code from consuming all the bytes from stdin and leaving
+            // no bytes for the actual ingest to read
+            val inputStream = new ByteArrayInputStream(inputBytes)
+            val iterator = CloseableIterator.single[InputStream](inputStream)
+            opened += iterator
+            System.setIn(inputStream)
+
+            // Return stdin
+            System.in
+          } catch {
+            case e: Exception => throw new RuntimeException("Unable to read from stdin", e)
+          }
+        }
+
+        open = openStdin
+      }
+      // try to infer the schema from the input files themselves
+      else {
+        Command.user.info("No converter defined - will attempt to detect schema from input files")
+        file = inputs.iterator.flatMap(PathUtils.interpretPath).headOption.getOrElse {
           throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
         }
+
+        def openFile(): InputStream = {
+          val streams = file.open.map(_._2)
+          opened += streams
+          if (streams.hasNext) {
+            streams.next
+          } else {
+            throw new ParameterException("Parameter <files> did not evaluate to anything that could be read")
+          }
+        }
+
+        open = openFile
       }
+
       val (inferredSft, inferredConverter) = try {
         val opt = format match {
-          case None      => SimpleFeatureConverter.infer(open, Option(sft), Option(file.path))
-          case Some(fmt) => TypeAwareInference.infer(fmt, open, Option(sft), Option(file.path))
+          case None => if (inputs == Inputs.StdInInputs) {
+              SimpleFeatureConverter.infer(open, Option(sft))
+            } else {
+              SimpleFeatureConverter.infer(open, Option(sft), Option(file.path))
+            }
+
+          case Some(fmt) => if (inputs == Inputs.StdInInputs) {
+            TypeAwareInference.infer(fmt, open, Option(sft), None)
+          } else {
+            TypeAwareInference.infer(fmt, open, Option(sft), Option(file.path))
+          }
```

Comment:
@elahrvivaz I made this change per your request, but the new unit tests will be unaffected because this code indirectly calls the `DelimitedTextConverterFactory.infer(is: InputStream, sft: Option[SimpleFeatureType], path: Option[String])` function, whose `path` argument is unused.

Are you referring to the `infer` functions from other converter classes (such as the `ParquetConverterFactory`) which use this `path` argument? In that case, do you think we need to add more unit tests to ensure this new logic works with other converters?

Reply:
right, I was referring to the other converter factory types. if you'd like to add tests for those, then feel free, but I won't hold up this PR for it. the type inference is currently only implemented for a subset of the converters, so the path may actually be unused in all of them, but we should still pass it through so that it's there if/when we implement additional type inferences that need it.

Reply:
Thank you for clarifying the details. Yes I agree - additional unit tests likely aren't a priority for this ticket, but I've made a new separate ticket for that work in case anyone wants to look into it: https://geomesa.atlassian.net/browse/GEOMESA-3328